### PR TITLE
RCAL-828: add PSF bits to source catalog.

### PR DESCRIPTION
**Pull request by mairanteodoro**
Resolves [RCAL-828](https://jira.stsci.edu/browse/RCAL-828)

This PR addresses the implementation of PSF photometry into `SourceCatalogStep`.

**Regression Tests**
Regression tests are all passing.
~https://plwishmaster.stsci.edu:8081/job/RT/job/Roman-Developers-Pull-Requests/775/~
~https://plwishmaster.stsci.edu:8081/job/RT/job/Roman-Developers-Pull-Requests/798/~
~https://plwishmaster.stsci.edu:8081/job/RT/job/Roman-Developers-Pull-Requests/808/~
~https://plwishmaster.stsci.edu:8081/job/RT/job/Roman-Developers-Pull-Requests/809/~
https://plwishmaster.stsci.edu:8081/job/RT/job/Roman-Developers-Pull-Requests/814/

**Checklist**
- [x] added entry in `CHANGES.rst` under the corresponding subsection
- [X] updated relevant tests
- [x] updated relevant documentation
- [ ] updated relevant milestone(s)
- [X] added relevant label(s)
- [X] ran regression tests, post a link to the Jenkins job below. [How to run regression tests on a PR](https://github.com/spacetelescope/romancal/wiki/Running-Regression-Tests-Against-PR-Branches)

### CODE REVIEW

**Review by schlafly — COMMENTED** [review]
This looks good to me.  Let's make sure it runs for both L2 and L3 products; I suspect it will not at the moment..  Can you compare a source_detection catalog with a source_catalog PSF catalog and see how similar they are?  We want the positions to be ~the same at the ~0.01 pix level.

File: romancal/source_catalog/source_catalog.py [hunk 1]
```diff
@@ -1073,6 +1085,119 @@ def _split_skycoord(self, table):
...
+        """
+        Perform PSF photometry by fitting PSF models to detected sources for refined astrometry.
+
+        This method constructs a gridded PSF model based on instrument and detector information.
+        It then fits the PSF model to the image model's sources to improve astrometric precision.
+
+        """
+        log.info("Constructing a gridded PSF model.")
+        filt = self.model.meta.instrument["optical_element"]
+        detector = self.model.meta.instrument["detector"].replace("WFI", "SCA")
```
[the remark this answers is not in the document]
Comment (mairanteodoro) [reply romancal/source_catalog/source_catalog.py:1171]: `source_catalog_model.meta` is defined in the `step.process()` call, and it is not passed on to the `RomanSourceCatalog` class instantiation. The current implementation just checks for the presence of `meta.instrument` in the datamodel during class instantiation.

**Comment by mairanteodoro** [comment]
Comparison between the results from `SourceDetectionStep` and `SourceCatalogStep`:
image

**Comment by mairanteodoro** [comment]
Working on updating the docs now.

File: romancal/source_catalog/source_catalog.py [hunk 2]
```diff
@@ -1073,6 +1085,118 @@ def _split_skycoord(self, table):
...
+            # MosaicModel (L3 datamodel)
+            filt = self.model.meta.basic.optical_element
+            detector = "SCA02"
+        gridded_psf_model, _ = psf.create_gridded_psf_model(
+            path_prefix="tmp",
+            filt=filt,
+            detector=detector,
+            overwrite=True,
+            logging_level="ERROR",
+        )
```
Comment (schlafly) [inline romancal/source_catalog/source_catalog.py:1179]: Dave reported an issue where the temporary gridded model was not cleaned up.  Can you fix this if it's still an issue, or create an issue to fix it in the future?

**Review by schlafly — COMMENTED** [review]
This looks good to me, thanks.  I left one inline comment.

File: romancal/source_catalog/source_catalog.py (hunk 2 above)
Comment (mairanteodoro) [reply romancal/source_catalog/source_catalog.py:1183]: The issue has been fixed.

**Review by schlafly — APPROVED** [review]

**Pull request merged**